### PR TITLE
[Issue-37] makefile: creating a Makefile

### PR DESCRIPTION
**Pull request by Bidon15**
The initial version contains such commands:
- help
- format
- lint
- test

### CODE REVIEW

**Comment by renaynay** [comment]
LGTM so far, let's maybe make an issue for future reference to add a `build` command (but this is dependent on us implementing the CLI, so might be a bit further down the line)

**Review by renaynay — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+format:
+	find . -name '*.go' -type f -not -path "*.git*" -not -name '*.pb.go' -not -name '*pb_test.go' | xargs gofmt -w -s
+	find . -name '*.go' -type f -not -path "*.git*"  -not -name '*.pb.go' -not -name '*pb_test.go' | xargs goimports -w -local github.com/celestiaorg/celestia-node
+.PHONY: format
+
+## lint: Linting *.go files using golangci-lint. Look for .golangci.yml for the list of linters.
+lint:
+	@echo "--> Running linter"
+	@golangci-lint run
+.PHONY: lint
```
Comment (Wondertan) [inline Makefile:20]: Don't think there is a need to keep `format`, as `lint` covers that.

**Review by Wondertan — COMMENTED** [review]
Forgot to submit the review and as Rene mentioned we would need a `build` command later

**Comment by Bidon15** [comment]
> LGTM so far, let's maybe make an issue for future reference to add a `build` command (but this is dependent on us implementing the CLI, so might be a bit further down the line)

Thank you for the review 🥳 
Yeap. Let's park it here: #43 

I'll add our implementation issue to the newly created issue, too

File: Makefile (hunk 1 above)
Comment (Bidon15) [reply Makefile:20]: OMT: I'm starting to go a little bit overboard, but having a super-duper combo like `make format && make lint && make test` as a one command might be helpful 🧐 wdyt?

Comment (marbar3778) [reply Makefile:20]: I think for now its fine as is. We dont have to make it perfect. A developer is free to use which command they want.

**Review by marbar3778 — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (Wondertan) [reply Makefile:20]: > While triggering gofmt, lint will only check if the file is formated, but not do the formating itself. It's one of the reason why I make the format as a separate command.

Oh, ok I missed that. Thanks!

**Review by Wondertan — APPROVED** [review]

**Comment by Wondertan** [comment]
@Bidon15, build failed for some reason

**Comment by Bidon15** [comment]
> @Bidon15, build failed for some reason

Thanks for the feedback @Wondertan 🎈 
Yes. It's expected as we still are developing for our awesome initial golang code-base 😸 
The errors are pointing to no golang code and absense of a module

**Comment by Wondertan** [comment]
@Bidon15, just tested `format` locally. Would be cool to prepend `@` to `find`, not print those long cmds every `format` execution.

**Pull request merged**